SRVOCF-302: Add docs on accessing secrets and ConfigMaps from Srvls functions (#34089)
openshift/openshift-docs: modules/serverless-functions-secrets-configmaps-manually.adoc

// Module included in the following assemblies:
//
// * serverless/functions/serverless-functions-accessing-secrets-configmaps.adoc

[id="serverless-functions-secrets-configmaps-manually_{context}"]
= Adding function access to secrets and config maps manually

You can manually add configuration for accessing secrets and config maps to your function.

.Procedure

. Open the `func.yaml` file for your function.

> . For each secret you want to mount as a volume, add the following YAML to the `volumes` section:

Reviewer: Can this be some level of heading (subsection for each example)? I am not sure if that's aligned with the rest of the docs. Something as I would do in markdown:

# Adding function access to secrets and config maps manually

You can manually add configuration for accessing secrets and config maps to your function.

## Mounting a secret as a volume
 For each secret you want to mount as a volume, add the following YAML to the `volumes` section:
...

## Mounting a config map as a volume
For each config map you want to mount as a volume, add the following YAML to the `volumes` section:
...

## Environment value from a key-value pair defined in a secret
...

...and so on.

Author: I preferred the single procedure way, but since you and Ashleigh are in favor of this, I went with the majority :)

+
[source,yaml]
----
name: test
namespace: ""
runtime: go
...
volumes:

> - secret: mysecret

Reviewer: I'd add an explanation to the fields as well, that under secret users should put the name of the secret and under path should be the path where the secret should be mounted.

  path: /workspace/secret
----

. For each config map you want to mount as a volume, add the following YAML to the `volumes` section:
+
[source,yaml]
----
name: test
namespace: ""
runtime: go
...
volumes:
- configMap: myconfigmap
  path: /workspace/configmap
----

> . For each value from a secret key-value pair that you want to assign to an environment variable, add the following YAML to the `envs` section:

Reviewer: I'd add a description for the fileds (what is the correct syntax for `value` field. Similar way (or a better one) as it is mentioned in upstream docs: odules/serverless-functions-secrets-configmaps-manually.adoc

The same for the other parts

// Module included in the following assemblies:
//
// * serverless/functions/serverless-functions-accessing-secrets-configmaps.adoc

[id="serverless-functions-secrets-configmaps-manually_{context}"]

> = Adding the configuration manually

Reviewer 2: Again, more descriptive heading would be good here, and this module needs an intro sentence to explain what it does / is used for

.Procedure

. Open the `func.yaml` file for your function.

> . For every secret you want to mount as a volume, add this to the `volumes` section:

Reviewer 2: ```suggestion
. For each secret you want to mount as a volume, add the following YAML to the `volumes` section:
```

+
[source,yaml]
----
- secret: mysecret
  path: /workspace/secret

> ----

Reviewer 2: I'd show the rest of the file to give more context rather than just the snippet, this doesn't show the `volumes` bit that's referred to or where in the file this appears

+

> For every ConfigMap you want to mount as a volume, add:

Reviewer 2: ```suggestion
For each config map you want to mount as a volume, add the following to the YAML file:
```
As above, make these consistent and show the full example YAML for context

+
[source,yaml]
----
- configMap: myconfigmap
  path: /workspace/configmap
----
+

> For every value from a secret key-value pair that you want to assign to an environment variable, add this to the `envs` section:

Reviewer 2: same comments as above for all the following steps

+
[source,yaml]
----
- name: EXAMPLE
  value: '{{ secret.mysecret.EXAMPLE }}'
----
+

> For every value from a ConfigMap key-value pair that you want to assign to an environment variable, add:

Reviewer 2: These all seem like they should be individual steps or procedures

+
[source,yaml]
----
- name: EXAMPLE
  value: '{{ configMap.myconfigmap.EXAMPLE }}'
----
+
For every secret, for which you want to import all key-value pairs as environment variables, add:
+
[source,yaml]
----
- value: '{{ secret.mysecret }}'
----
+
For every ConfigMap, for which you want to import all key-value pairs as environment variables, add:
+
[source,yaml]
----
- value: '{{ configMap.myconfigmap }}'
----
+

> An example of the resulting function configuration might be:

Reviewer 2: ```suggestion
The resulting function configuration will look similar to the following YAML:
```
suggestion

+

> .Example

Reviewer 2: ```suggestion
.Example function configuration
```

[source,yaml]
----
name: test
namespace: ""
runtime: go
...
volumes:
- secret: mysecret
  path: /workspace/secret
- configMap: myconfigmap
  path: /workspace/configmap
envs:
- name: SECRET2
  value: '{{ secret.mysecret2.SECRET2 }}'
- value: '{{ secret.mysecret3 }}'
- name: CONFIGMAP2
  value: '{{ configMap.myconfigmap2.CONFIGMAP2 }}'
- value: '{{ configMap.myconfigmap3 }}'
----

> . Save the configuration and, optionally, deploy the function:

Reviewer 2: make this two steps and follow the guidelines wrt optional steps

+
[source,terminal]
----
$ kn func deploy -p test
----
+

> If a secret or ConfigMap value cannot be accessed, the deployment fails with an error message specifying the inaccessible values.

Reviewer 2: same comments as above for this one
